bigtable/cmd/cbt/cbt_test.go

```diff
@@ -171,3 +178,401 @@ func TestParseColumnsFilter(t *testing.T) {
 		}
 	}
 }
+
+func TestCsvImporterArgs(t *testing.T) {
+	tests := []struct {
+		in   []string
+		out  importerArgs
+		fail bool
+	}{
+		{in: []string{"my-table", "my-file.csv"}, out: importerArgs{"", []string{""}, 500, 1}},
+		{in: []string{"my-table", "my-file.csv", "app-profile="}, out: importerArgs{"", []string{""}, 500, 1}},
+		{in: []string{"my-table", "my-file.csv", "column-family="}, out: importerArgs{"", []string{"", ""}, 500, 1}},
+		{in: []string{"my-table", "my-file.csv", "app-profile=my-ap", "column-family=my-family", "batch-size=100", "workers=20"},
+			out: importerArgs{"my-ap", []string{"", "my-family"}, 100, 20}},
+
+		{in: []string{}, fail: true},
+		{in: []string{"my-table", "my-file.csv", "batch-size=-5"}, fail: true},
+		{in: []string{"my-table", "my-file.csv", "batch-size=5000000"}, fail: true},
+		{in: []string{"my-table", "my-file.csv", "batch-size=nan"}, fail: true},
+		{in: []string{"my-table", "my-file.csv", "batch-size="}, fail: true},
+		{in: []string{"my-table", "my-file.csv", "workers=0"}, fail: true},
+		{in: []string{"my-table", "my-file.csv", "workers=nan"}, fail: true},
+		{in: []string{"my-table", "my-file.csv", "workers="}, fail: true},
+	}
+	for _, tc := range tests {
+		got, err := parseImporterArgs(context.Background(), tc.in)
+		if !tc.fail && err != nil {
+			t.Errorf("parseImportArgs(%q) unexpectedly failed: %v", tc.in, err)
+			continue
+		}
+		if tc.fail && err == nil {
+			t.Errorf("parseImportArgs(%q) did not fail, out: %q", tc.in, got)
+			continue
+		}
+		if tc.fail {
+			continue
+		}
+		if got.appProfile != tc.out.appProfile ||
+			len(got.fams) != len(tc.out.fams) ||
+			got.sz != tc.out.sz ||
+			got.workers != tc.out.workers {
+			t.Errorf("parseImportArgs(%q) did not fail, out: %q", tc.in, got)
+			continue
+		}
+		for i, f := range got.fams {
+			if f != tc.out.fams[i] {
+				t.Errorf("parseImportArgs(%q) incorrect column families, out: %q", tc.in, got)
+				continue
+			}
+		}
+	}
+}
+
+func writeAsCSV(data [][]string) ([]byte, error) {
```

Comment:
at first I thought this was a mock, maybe we should add a comment that this is to transform test data for the test?

Reply:
reworded to "transformToCsvBuffer()" to be more descriptive, can add more comments if required